frontend/javascripts/router.js

```diff
@@ -428,6 +433,24 @@ class ReactRouter extends React.Component<Props> {
                   />
                 )}
               />
+              <Route
+                path="/datasets/:organizationName/:dataSetName/createExplorative/:type/:withFallback"
+                render={({ match, location }: ContextRouter) => (
+                  <AsyncRedirect
+                    redirectTo={async () => {
+                      const dataset = {
```

Comment:
I think, you should be able to add something like this:

```
if (!match.params.organizationName || !match.params.dataSetName || ...) {
  // Typehint for flow
  throw new Exception("Invalid URL");
}
```

The exception should never occur since the router wouldn't execute that branch, but flow doesn't know that.